tidy3d/components/data/data_array.py

```diff
@@ -283,6 +284,37 @@ def multiply_at(self, value: complex, coord_name: str, indices: List[int]) -> Da
         self_mult[{coord_name: indices}] *= value
         return self_mult
 
+    def interp(
+        self,
+        coords: Mapping[Any, Any] | None = None,
+        method: InterpOptions = "linear",
+        assume_sorted: bool = False,
+        kwargs: Mapping[str, Any] | None = None,
+        **coords_kwargs: Any,
```

Comment:
are we safe if the xarray API changes in the future? (or if it's different in the past). or will it break this?

Reply:
I copied these arguments directly from xarray. If they change the API then this will likely break, but then those xarray version wouldn't be compatible anyways?